core/src/main/java/com/yahoo/oak/AbstractSlice.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+/* An abstract Slice represents the information about an off-heap cut: a portion of a bigger block,
+** which is part of the underlying managed off-heap memory. Concrete implementation adapts
+** the abstract Slice to a specific memory manager.
+** Slice is allocated only via memory manager, and can be de-allocated later.
+** Slice can be either empty or associated with an off-heap cut,
+** which is the aforementioned portion of an off-heap memory.*/
+abstract class AbstractSlice implements Slice, Comparable<AbstractSlice> {
```

Comment:
Maybe we should call it `BlockAllocationSlice`?

Reply:
OK